docs/utxo-hd/v0.1/for-developers.md

```diff
@@ -0,0 +1,456 @@
+    current --> emptyInputState
+    classDef operation fill:#BDB76B
+    state "seqDiff :: LedgerTables l SeqDiffMK" as seqDiff
+    getDiffs:::dbchangelog --> seqDiff
+    seqDiff --> forwarding:::operation
+    unforwardedTables --> forwarding
+    state "forwardedTables :: LedgerTables l ValuesMK" as forwardedTables
+    forwarding --> forwardedTables
+    emptyInputState --> withLedgerTables:::operation
+    forwardedTables --> withLedgerTables
+    withLedgerTables --> inputState
+    state "inputState :: l ValuesMK" as inputState
+    inputState --> stow:::operation
+    state "stowedState :: l EmptyMK" as stowedState
+    stow --> stowedState
+    block --> apply:::operation
+    stowedState --> apply
+    state "resultingState :: l EmptyMK" as resultingState
+    apply --> resultingState
+    resultingState --> unstow:::operation
+    state "unstowedState :: l ValuesMK" as unstowedState
+    unstow --> unstowedState
+    unstowedState --> projectLedgerTables:::operation
+    state "resultingTables :: LedgerTables l ValuesMK" as resultingTables 
+    projectLedgerTables --> resultingTables
+    unstowedState --> forgetLedgerTables:::operation
+    state "emptyResultingState :: l EmptyMK" as emptyResultingState
+    forgetLedgerTables --> emptyResultingState 
+    state "dbChangelog" as dbChangelog2
+    emptyResultingState --> withLedgerTables2
+    resultingTables --> diff:::operation
+    forwardedTables --> diff
+    state "diffTables :: LedgerTables l DiffMK" as diffTables
+    diff --> diffTables
+    state "withLedgerTables" as withLedgerTables2
+    diffTables --> withLedgerTables2:::operation
+    state "diffState :: l DiffMK" as diffState
+    withLedgerTables2 --> diffState
+    diffState --> dbChangelog2:::ledgerdb: to store
+```
+
+### Era boundaries
+
+In general, ticking a ledger state should not modify the values in the UTxO set,
+but there are two exceptions in the Cardano case when ticking crosses an era
+boundary. For this reason, the general scheme for **ticking a block is: a ledger
+state without a UTxO set (`EmptyMK`) goes in and a ledger state with some deltas
+(`DiffMK`) comes out**. This is reflected in `applyChainTickLedgerResult`.
+
+```haskell
+applyChainTickLedgerResult ::
+     LedgerCfg l
+  -> SlotNo
+  -> l EmptyMK
+  -> LedgerResult l (Ticked1 l DiffMK)
+```
+
+### Ticking and era boundaries
+
+In general, ticking a ledger state won't modify the values in the UTxO set, but
+there are two exceptions in the Cardano case when ticking crosses an era
+boundary: 
+
+- For simplicity, we have deliberately chosen that the Byron era would not have
+  any ledger tables, and there would only be ledger tables starting in Shelley.
+  This means that when crossing the boundary to Shelley, the whole UTxO set will
+  be returned as deltas (see `translateLedgerStateByronToShelleyWrapper`
+  [here](../../../ouroboros-consensus-cardano/src/ouroboros-consensus-cardano/Ouroboros/Consensus/Cardano/CanHardFork.hs)).
```

Comment:
It might be worth mentioning that this UTxO set was small compared to today's UTxO. (... it was, right?)

Reply:
Done